File: cpp/4/10810.cpp
#include <iostream>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    int baguni[101] = {0};

    for (int i = 0; i < m; i++) {
        int x, y, z;
        cin >> x >> y >> z;
        for (int j = x; j <= y; j++) {
            baguni[j] = z;
        }
    }

    for (int i = 1; i <= n; i++) {
        cout << baguni[i] << " ";
    }

    return 0;
}

File: kotlin/4/10810.kt
fun main() {
    var x = readLine()!!.split(" ").map { it.toInt() }
    var n = x[0]
    var m = x[1]
    var baguni = IntArray(n)

    for (o in 0 until(m)) {
        var y = readLine()!!.split(" ").map { it.toInt() }
        val z = y[0]
        val t = y[1]
        val s = y[2]
        
        for (i: Int in z..t) {
            baguni[i - 1] = s
        }
    }

    for (i in baguni) {
        print("${i} ")
    }
}